### ISSUE TITLE
Coq exists because of bad python version

### ISSUE THREAD

**Issue by abialystok** [ms-jpq/coq_nvim#461]
The python version in my PATH is 3.10.2:
```
~ python --version
Python 3.10.2
```

However when I start Coq it says:
```
⛔️ python < 3.8.2
COQ EXITED - 1
```
Happens on Ubuntu 18.04.6, neovim version 0.6.1

**Comment by abialystok** [comment]
Got it to work - it turns out that Coq uses `python3`, which for me was pointing at `python3.6` so it wasn't good.
Then, after I fixed that and tried to run `COQdeps` I encountered the message "Please install python3-venv separately. (apt, yum, apk, etc)", although venv was already installed.

When I printed the exception that caused this message to appear, I saw that what actually failed was the command `python3 -Im ensurepip --default-pip`. This happened because I did not have `ensurepip` installed. 

The reason `ensurepip` wasn't install was because I downloaded python via deadsnakes (see https://github.com/deadsnakes/issues/issues/79). As stated in the above issue, if you install venv from deadsnakes instead of the default apt repository, it comes with `ensurepip`. I did that by running `apt install python3.10-venv` instead of `apt install python3-venv`.

Long story short - now everything is working for me. Figured I'd right my journey down in case anyone would bump into one of these issues.

**Comment by opentyler** [comment]
Just wanted to say thanks for writing out your solution! I just ran into the ensurepip missing error and was searching high and low for what was going on. Your "apt install python3.10-venv" was the solution I was looking for, thank you!

**Comment by Watynecc** [comment]
Thanks dood It was just redoing :COQdeps after installing `python39-tools` on opensuse leap !
